bitnami/elasticsearch/templates/data/statefulset.yaml

```
@@ -85,7 +85,7 @@ spec:
       {{- end }}
       {{- if or (or .Values.initContainers .Values.data.initContainers) .Values.security.enabled .Values.sysctlImage.enabled (and .Values.volumePermissions.enabled .Values.data.persistence.enabled) }}
       initContainers:
-        {{- if .Values.sysctlImage.enabled }}
+        {{- if and .Values.sysctlImage.enabled (default true .Values.enableDefaultInitContainers) }}
```

By default `enableDefaultInitContainers` is set to `true`, you can simplify the condition removing the default here: 

```suggestion
        {{- if and .Values.sysctlImage.enabled .Values.enableDefaultInitContainers }}
```

This suggestion applies also to the other modifications.